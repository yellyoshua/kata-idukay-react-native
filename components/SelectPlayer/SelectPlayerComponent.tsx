import React from 'react';
import { StyleSheet, TouchableOpacity, View, Text, Image, ImageSourcePropType } from 'react-native';
import { HechiceroProps } from "../../types";

export default function SelectPlayer({ setPlayer }: { setPlayer: (val: HechiceroProps) => void }) {

  return <View style={styles.container}>
    <Text style={styles.title}>
      Select your player
    </Text>
    <View style={styles.row}>
      {
        [
          { alias: "smurf", name: "Smurf", avatar: "https://raw.githubusercontent.com/yellyoshua/kata-idukay-react/master/src/icons/avatars/smurf.png" },
          { alias: "thor", name: "Thor", avatar: "https://raw.githubusercontent.com/yellyoshua/kata-idukay-react/master/src/icons/avatars/thor.png" },
          { alias: "wonder-woman", name: "Wonder Woman", avatar: "https://raw.githubusercontent.com/yellyoshua/kata-idukay-react/master/src/icons/avatars/wonder-woman.png" },
          { alias: "wizard-original", name: "Wizard Original", avatar: "https://raw.githubusercontent.com/yellyoshua/kata-idukay-react/master/src/icons/avatars/wizard-original.png" },
        ].map((player: HechiceroProps, index: number) => (
          <PlayerCard key={index} player={player} onPressed={setPlayer} />
        ))
      }
    </View>
  </View>
}

export function PlayerCard({ player, onPressed }: { player: HechiceroProps; onPressed?: (player: HechiceroProps) => void }) {
  return <TouchableOpacity style={styles.playerCard} onPress={() => {
    if (onPressed) {
      onPressed(player)
    }
  }}>
    <Image style={styles.playerCardImage} resizeMode="cover" source={{ uri: player.avatar }} />
    <Text style={styles.playerCardTitle}>{player.name}</Text>
  </TouchableOpacity>
}

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center'
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold',
    color: "#fff",
    textAlign: "center"
  },
  row: {
    flex: 1,
    justifyContent: "center",
    flexDirection: "row",
    flexWrap: "wrap"
  },
  playerCard: {
    width: 150,
    margin: 10,
    padding: 10,
    backgroundColor: "#33363e",
    borderRadius: 10,
    textAlign: "center"
  },
  playerCardImage: {
    width: 120,
    height: 120,
    marginHorizontal: "auto"
  },
  playerCardTitle: {
    fontSize: 18,
    color: "#fff",
    fontWeight: "bold",
    paddingVertical: 10
  }
});