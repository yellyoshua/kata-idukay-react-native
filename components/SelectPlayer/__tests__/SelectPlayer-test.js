import * as React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import SelectPlayer from '../SelectPlayerComponent';

describe("unit test select-player component",() => {

  it(`renders correctly`, () => {
    const setPlayer = jest.fn();
    const player = renderer.create(<SelectPlayer setPlayer={setPlayer}/>).toJSON();

    expect(player).toMatchSnapshot();
  });

  it("Be rendered all players",() => {
    const setPlayer = jest.fn();
    const {getByText} = render(<SelectPlayer setPlayer={setPlayer}/>);
  
    expect(getByText("Thor")).toBeTruthy();
    expect(getByText("Wonder Woman")).toBeTruthy();
    expect(getByText("Smurf")).toBeTruthy();
    expect(getByText("Wizard Original")).toBeTruthy();
  });

  it(`Thor be selected`, () => {
    const setPlayer = jest.fn();
    const {getByText} = render(<SelectPlayer setPlayer={setPlayer}/>);
  
    fireEvent.press(getByText("Thor"));
  
    expect(setPlayer).toBeCalledWith({
      alias: "thor",
      avatar: "https://raw.githubusercontent.com/yellyoshua/kata-idukay-react/master/src/icons/avatars/thor.png",
      name: "Thor"
    });
  });

  it(`Wonder Woman be selected`, () => {
    const setPlayer = jest.fn();
    const {getByText} = render(<SelectPlayer setPlayer={setPlayer}/>);
  
    fireEvent.press(getByText("Wonder Woman"));
  
    expect(setPlayer).toBeCalledWith({
      alias: "wonder-woman",
      avatar: "https://raw.githubusercontent.com/yellyoshua/kata-idukay-react/master/src/icons/avatars/wonder-woman.png",
      name: "Wonder Woman"
    });
  });

  it(`Smurf be selected`, () => {
    const setPlayer = jest.fn();
    const {getByText} = render(<SelectPlayer setPlayer={setPlayer}/>);
  
    fireEvent.press(getByText("Smurf"));
  
    expect(setPlayer).toBeCalledWith({
      alias: "smurf",
      avatar: "https://raw.githubusercontent.com/yellyoshua/kata-idukay-react/master/src/icons/avatars/smurf.png",
      name: "Smurf"
    });
  });

  it(`Wizard Original be selected`, () => {
    const setPlayer = jest.fn();
    const {getByText} = render(<SelectPlayer setPlayer={setPlayer}/>);
  
    fireEvent.press(getByText("Wizard Original"));
  
    expect(setPlayer).toBeCalledWith({
      alias: "wizard-original",
      avatar: "https://raw.githubusercontent.com/yellyoshua/kata-idukay-react/master/src/icons/avatars/wizard-original.png",
      name: "Wizard Original"
    });
  });
})
