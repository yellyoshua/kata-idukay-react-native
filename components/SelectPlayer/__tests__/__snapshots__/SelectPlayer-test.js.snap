// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit test select-player component renders correctly 1`] = `
<View
  style={
    Object {
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 20,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Select your player
  </Text>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#33363e",
          "borderRadius": 10,
          "margin": 10,
          "opacity": 1,
          "padding": 10,
          "textAlign": "center",
          "width": 150,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "uri": "https://raw.githubusercontent.com/yellyoshua/kata-idukay-react/master/src/icons/avatars/smurf.png",
          }
        }
        style={
          Object {
            "height": 120,
            "marginHorizontal": "auto",
            "width": 120,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 18,
            "fontWeight": "bold",
            "paddingVertical": 10,
          }
        }
      >
        Smurf
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#33363e",
          "borderRadius": 10,
          "margin": 10,
          "opacity": 1,
          "padding": 10,
          "textAlign": "center",
          "width": 150,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "uri": "https://raw.githubusercontent.com/yellyoshua/kata-idukay-react/master/src/icons/avatars/thor.png",
          }
        }
        style={
          Object {
            "height": 120,
            "marginHorizontal": "auto",
            "width": 120,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 18,
            "fontWeight": "bold",
            "paddingVertical": 10,
          }
        }
      >
        Thor
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#33363e",
          "borderRadius": 10,
          "margin": 10,
          "opacity": 1,
          "padding": 10,
          "textAlign": "center",
          "width": 150,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "uri": "https://raw.githubusercontent.com/yellyoshua/kata-idukay-react/master/src/icons/avatars/wonder-woman.png",
          }
        }
        style={
          Object {
            "height": 120,
            "marginHorizontal": "auto",
            "width": 120,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 18,
            "fontWeight": "bold",
            "paddingVertical": 10,
          }
        }
      >
        Wonder Woman
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#33363e",
          "borderRadius": 10,
          "margin": 10,
          "opacity": 1,
          "padding": 10,
          "textAlign": "center",
          "width": 150,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "uri": "https://raw.githubusercontent.com/yellyoshua/kata-idukay-react/master/src/icons/avatars/wizard-original.png",
          }
        }
        style={
          Object {
            "height": 120,
            "marginHorizontal": "auto",
            "width": 120,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 18,
            "fontWeight": "bold",
            "paddingVertical": 10,
          }
        }
      >
        Wizard Original
      </Text>
    </View>
  </View>
</View>
`;
